### PR TITLE
Add Operations in REST API And Use Templated Call With Retry

### PR DESCRIPTION
**Pull request by niruhan**
## Purpose
Add the following operations:
* getDeleted
* getUpdated
* setPassword

Make the following changes related to calling endpoints
* Change access token refreshing from expiry based to retry based
* Use class mediator for access token refreshing
* template the endpoint calling with different http methods, blocking and non-blocking, and retry if 4XX returned
* remove call mediator usage in all the method template and use templated call

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: src/main/resources/salesforcerest-records/getDeleted.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                            <markForSuspension>
+                                <errorCodes>-1</errorCodes>
+                            </markForSuspension>
+                        </http>
+                    </endpoint>
+                </call>
+            </else>
+        </filter>
+    </sequence>
+</template>
```
Comment (dinuish94) [inline src/main/resources/salesforcerest-records/getDeleted.xml]: Add new line at the end of the file

File: src/main/resources/salesforcerest-records/getUpdated.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                            <markForSuspension>
+                                <errorCodes>-1</errorCodes>
+                            </markForSuspension>
+                        </http>
+                    </endpoint>
+                </call>
+            </else>
+        </filter>
+    </sequence>
+</template>
```
Comment (dinuish94) [inline src/main/resources/salesforcerest-records/getUpdated.xml]: Add new line at the end of the file

File: src/main/resources/salesforcerest-users/setPassword.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                            <markForSuspension>
+                                <errorCodes>-1</errorCodes>
+                            </markForSuspension>
+                        </http>
+                    </endpoint>
+                </call>
+            </else>
+        </filter>
+    </sequence>
+</template>
```
Comment (dinuish94) [inline src/main/resources/salesforcerest-users/setPassword.xml]: Add new line at the end of the file

File: src/main/resources/salesforcerest-users/component.xml [hunk 4]
```diff
@@ -26,5 +26,9 @@
             <file>resetPassword.xml</file>
             <description>Reset Password for Salesforce account for a specific User.</description>
         </component>
+        <component name="setPassword">
+            <file>setPassword.xml</file>
+            <description>Set new password for Salesforce account for a specific User.</description>
+        </component>
     </subComponents>
 </component>
```
Comment (dinuish94) [inline src/main/resources/salesforcerest-users/component.xml]: Add new line at the end of the file

File: src/main/resources/salesforcerest-records/getDeleted.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Vathsan) [inline src/main/resources/salesforcerest-records/getDeleted.xml]: Shall we update the license?

**Review by abeykoon — APPROVED** [review]

**Pull request merged**